go/signature/signature_key_templates_test.go

```diff
@@ -19,65 +19,141 @@ import (
+		curveType   commonpb.EllipticCurveType
+		hashType    commonpb.HashType
+		sigEncoding ecdsapb.EcdsaSignatureEncoding
+		prefixType  tinkpb.OutputPrefixType
+	}{
+		{
+			tcName:      "test ECDSA P-256 with SHA256, DER format and TINK output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP256KeyTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P256,
+			hashType:    commonpb.HashType_SHA256,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "test ECDSA P-384 with SHA512, DER format and TINK output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP384KeyTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P384,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "test ECDSA P-521 with SHA512, DER format and TINK output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP521KeyTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P521,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "test ECDSA P-256 with SHA256, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP256KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P256,
+			hashType:    commonpb.HashType_SHA256,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "test ECDSA P-384 with SHA512, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP384KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P384,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "test ECDSA P-521 with SHA512, DER format and RAW output prefix",
+			typeURL:     testutil.ECDSASignerTypeURL,
+			sigTemplate: signature.ECDSAP521KeyWithoutPrefixTemplate(),
+			curveType:   commonpb.EllipticCurveType_NIST_P521,
+			hashType:    commonpb.HashType_SHA512,
+			sigEncoding: ecdsapb.EcdsaSignatureEncoding_DER,
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
+		{
+			tcName:      "test ED25519 with TINK output prefix",
+			typeURL:     testutil.ED25519SignerTypeURL,
+			sigTemplate: signature.ED25519KeyTemplate(),
+			prefixType:  tinkpb.OutputPrefixType_TINK,
+		},
+		{
+			tcName:      "test ED25519 with RAW output prefix",
+			typeURL:     testutil.ED25519SignerTypeURL,
+			sigTemplate: signature.ED25519KeyWithoutPrefixTemplate(),
+			prefixType:  tinkpb.OutputPrefixType_RAW,
+		},
```

Comment:
Please add a new test for ED25519 key templates. They aren't different from ECDSA key templates.

Reply:
The two elements above are for ED25519 tests. I'm not sure I understand the request.

Reply:
I see what you mean, I need to create another Test function for EDDSA, similar to ECDSA.

I will split the functions in a bit.

Reply:
created TestEDDSAKeyTemplates() for ED25519 key templates

Reply:
Renamed function to `TestED25519KeyTemplates()` instead.